mubench.pipeline/tasks/implementations/compile.py

```diff
@@ -158,24 +158,56 @@ def _compile(commands: List[str], project_dir: str, dep_dir: str) -> None:
         for command in commands:
             if command.startswith("mvn "):
                 command = "mvn dependency:build-classpath " + command[4:]
-                output = Shell.exec(command, cwd=project_dir,logger=logger)
+                output = Shell.exec(command, cwd=project_dir, logger=logger)
                 dependencies = Compile._parse_maven_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             elif command.startswith("ant"):
                 command += " -debug -verbose"
-                output = Shell.exec(command, cwd=project_dir,logger=logger)
+                output = Shell.exec(command, cwd=project_dir, logger=logger)
                 dependencies = Compile._parse_ant_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             elif command.startswith("gradle "):
-                Shell.exec(command,cwd=project_dir,logger=logger)
-                shutil.copy(os.path.dirname(__file__) + '/classpath.gradle', project_dir)
-                output = Shell.exec("gradle :printClasspath -b classpath.gradle", cwd=project_dir, logger=logger)
+                Shell.exec(command, cwd=project_dir, logger=logger)
+                buildfile_dir, classpath_cmd = Compile._parse_buildfile_dir(command)
+                shutil.copy(os.path.dirname(__file__) + '/classpath.gradle', project_dir + "/" + buildfile_dir)
+                output = Shell.exec(classpath_cmd, cwd=project_dir, logger=logger)
                 dependencies = Compile._parse_gradle_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             else:
                 Shell.exec(command, cwd=project_dir, logger=logger)
 
     @staticmethod
+    def _parse_buildfile_dir(command):
```

Comment:
I expected this method to only parse the dir, not to also construct the classpath_command. Create the command string in the caller (like you do for Maven and Ant), then this method returns only a string, instead of a tuple.

Btw. the parse_... methods can all be private (i.e., start with two _), right?